src/s1reader/s1_reader.py

```diff
@@ -279,8 +285,89 @@ def get_ipf_version(tree: ET):
 
     return ipf_version
 
+def get_path_aux_cal(directory_aux_cal: str, str_annotation: str):
+    '''
+    Decide which aux_cal to load
+    Criteria based on ESA document in the link below:
+    https://sentinel.esa.int/documents/247904/1877131/Sentinel-1_IPF_Auxiliary_Product_Specification
+
+    Parameters:
+    -----------
+    diectory_aux_cal: str
+        Directory for the AUX_CAL .zip files
+    str_annotation: str
+        annotation_path that is used in `burst_from_xml()`
+
+    Return:
+    --------
+    path_aux_cal: str
+        Path to the AUX_CAL file that corresponds to the criteria provided
+        None if the matching AUX_CAL is not found in `directory_aux_cal`
+
+    '''
+
+    # extract the date string and platform info from str_annotation
+    str_safe = os.path.basename(str_annotation.split('.SAFE')[0])
+
+    token_safe = str_safe.split('_')
+    str_platform = token_safe[0]
+    str_sensing_start = token_safe[5]
+
+    list_aux_cal = glob.glob(f'{directory_aux_cal}/{str_platform}_AUX_CAL_V*.SAFE.zip')
+
+    if len(list_aux_cal) == 0:
+        raise ValueError( 'Cannot find AUX_CAL files from directory: '
+                                f'{directory_aux_cal}')
+
+    format_datetime = '%Y%m%dT%H%M%S'
+
+    datetime_sensing_start = datetime.datetime.strptime(str_sensing_start,format_datetime)
+
+    # sequentially parse the time info of AUX_CAL and search for the matching file
+    id_match = None
+    dt_validation_prev = None
+    dt_generation_prev = 1  # dummy value
+    for i, path_aux_cal in enumerate(list_aux_cal):
+        token_aux_cal = os.path.basename(path_aux_cal).split('_')
+        datetime_validation = datetime.datetime.strptime(token_aux_cal[3][1:],
+                                                         format_datetime)
+        datetime_generation = datetime.datetime.strptime(token_aux_cal[4][1:].split('.')[0],
+                                                         format_datetime)
+
+        dt_validation = int((datetime_sensing_start - datetime_validation).total_seconds())
+        dt_generation = int((datetime_sensing_start - datetime_generation).total_seconds())
+
+        if dt_validation>=0:
+            if dt_validation_prev is None:
+                # Initial allocation
+                id_match = i
+                dt_validation_prev = dt_validation
+                dt_generation_prev = dt_generation
+                continue
+
+            if dt_validation_prev > dt_validation:
+                id_match = i
+                dt_validation_prev = dt_validation
+                dt_generation_prev = dt_generation
+                continue
+
+            if dt_validation_prev == dt_validation:
+                # Same validity time; Choose the one with latest generation time
+                if dt_generation_prev > dt_generation:
+                    id_match = i
+                    dt_generation_prev = dt_generation
```

Comment:
Thank you @seongsujeong for the updates. Sorry, I don't understand what the rule is here for selecting the AUX_CAL file. I thought that maybe we wanted to (1) compare when the dataset was acquired and compare with the "validation date" first, and (2) if we don't find a match, we'd take one of the AUX_CAL files that are available for that configuration ID. Maybe you are doing the latter here (2) by getting the AUX_CAL with latest "validation date", but I don't see the former (1).

Reply:
I've took some time to search through the related document and was able to found the selection criteria suggested by ESA, which is:

>The criteria for selecting an appropriate auxiliary product to use when processing a particular job order are as follows:
>1. Select the auxiliary product(s) with a validity start date/time closest to, but not later than, the start of the job order;
>2. If there is more than one product which meets the first criteria (e.g. two auxiliary files have the same validity date/time), then use the auxiliary product with the latest generation time.

The revised function is the implementation of the criteria above.
The link to the document is provided in the docstring.

Reply:
That's great that you found this, @seongsujeong . It looks that your code is prioritizing the latest `dt_validation_prev`, which doesn't seem to be what they suggest? Or maybe I'm misunderstanding this. If everything is correct, could you please add some comments so it's easier to understand the code? 
Tagging @vbrancat and @LiangJYu so they can follow this discussion.

Reply:
`dt_validation_prev` in fact the "best `dt_validation`" that satisfies the 1st criteria above.
The algorithm here is to iterate through the found AUX_CAL files, and compare the `dt_validation` in the current round with the "most suitable" validation time that has found so far (i.e. dt_validation_prev). `dt_validation_prev` gets updated when `dt_validation` is better (and the iteration goes on).

Maybe changing `dt_validation_prev` as sth. like `dt_validation_best_so_far` would cause less confusion?

Reply:
You are right, @seongsujeong . Thank you for all info! Would you please add the text by ESA in the comments? I think that will be very helpful to anyone reading the code. 

For RTC, I'd be a little worried about the case that you don't find a suitable AUX_CAL and the exception was raised for a correction that is not needed (at least for RTC), but I'm glad that you are adding a flag to turn it off, so this will not be a problem. Thanks! 

Also, please "early escape" the `if` statement `dt_validation<0` with a `continue`.

Reply:
Sounds good. I've applied early escape, and added more information on  the docstring and the comments between the lines to help in better understanding the algorithm.